api/lib/gateways/Academy-Benefits/Search.js

```diff
@@ -1,82 +1,90 @@
+const rp = require('request-promise');
+const { Systems } = require('../../Constants');
 const moment = require('moment');
 
 module.exports = options => {
-  const searchDb = options.searchDb;
-  const searchAPI = options.searchAPI;
   const logger = options.logger;
+  const baseUrl = options.baseUrl;
+  const apiKey = options.apiKey;
+  const buildSearchRecord = options.buildSearchRecord;
 
-  const getDbRecords = async queryParams => {
-    try {
-      return await searchDb.execute(queryParams);
-    } catch (err) {
-      logger.error(
-        `Error searching customers in Academy-Benefits: ${err}`,
-        err
-      );
-      return [];
+  const search = async queryParams => {
+    let response = await callApi(queryParams);
+    let claimants = response.claimants;
+
+    while (response.nextCursor) {
+      response = await callApi(queryParams, response.nextCursor);
+      claimants = [...claimants, ...response.claimants];
     }
+    return processRecords(claimants);
   };
 
   const getApiRecords = async queryParams => {
     try {
-      return await searchAPI.execute(queryParams);
+      return await search(queryParams);
     } catch (err) {
       logger.error(
-        `Error searching customers in Academy-Benefits API: ${err}`,
+        `Error searching customers in Academy-Benefits API: ${err.error}`,
         err
       );
       return [];
     }
   };
 
-  const checkRecordsAreIdentical = (dbRecords, apiRecords) => {
```

Comment:
Do we no longer need to validate that the records are identical?

Reply:
We are now only retrieving records from the API and then returning them. So we won't call the database at all.